juriscraper/opinions/united_states/state/nyappterm_1st.py

```diff
@@ -45,148 +31,32 @@ def __init__(self, *args, **kwargs):
-
-        # use selenium to establish required cookies
-        logger.info("Running Selenium browser to get the cookies...")
-        add_delay(20, 5)
-        self.initiate_webdriven_session()
-        logger.info(f"Using cookies: {self.cookies}")
-        request_dict.update({"cookies": self.cookies})
-
-        html__ = super()._download(request_dict)
-        i = 0
-        while not html__.xpath("//table") and i < 10:
-            add_delay(20, 5)
-            html__ = super()._download(request_dict)
-            i += 1
-            logger.info(f"Got a bad response {i} time(s)")
-        return html__
-
-    def _get_case_names(self):
-        case_names = []
-        for element in self.html.xpath(self.base_path):
-            case_names.append(
-                "".join(x.strip() for x in element.xpath("./td[1]//text()"))
-            )
-        return case_names
-
-    def _get_download_urls(self):
-        download_urls = []
-        for element in self.html.xpath(self.base_path):
-            url = ""
-            for href in element.xpath("./td[5]//@href"):
-                href = clean_if_py3(href)
-                # Check for newer standard href
-                match = self.href_standard.match(href)
-                if match:
-                    url = match.group(0)
-                    break
-                # Check for presence of legacy JavaScript href
-                matches = self.href_js.findall(href)
-                if matches:
-                    url = url + matches[0]
-            if url:
-                download_urls.append(url)
-        return download_urls
-
-    def _get_case_dates(self):
-        case_dates = []
-        for element in self.html.xpath(f"{self.base_path}/td[2]//text()"):
-            case_dates.append(datetime.strptime(element.strip(), "%m/%d/%Y"))
-        return case_dates
-
-    def _get_precedential_statuses(self):
-        return ["Published"] * len(self.case_names)
-
-    def _get_docket_numbers(self):
-        docket_numbers = []
-        for element in self.html.xpath(self.base_path):
-            docket_numbers.append(
-                "".join(x.strip() for x in element.xpath("./td[5]//text()"))
+    def _process_html(self):
+        for row in self.html.xpath(".//table")[-1].xpath(".//tr")[1:]:
+            docket = " ".join(row.xpath("./td[5]//text()"))
+            citation = " ".join(row.xpath("./td[4]//text()"))
+            url = row.xpath(".//a")[0].get("href")
+            url = re.findall(r"(http.*htm)", url)[0]
+            self.cases.append(
+                {
+                    "name": row.xpath(".//td")[0].text_content(),
+                    "date": row.xpath(".//td")[1].text_content(),
+                    "url": url,
+                    "status": "Published"
+                    if "(U)" not in docket
+                    else "Unpublished",
+                    "docket": docket,
+                    "neutral_citation": docket,
```

Comment:
They're doing something a bit weird here, I think. It seems like their docket numbers aren't provided in the search results page and you have to click through to the opinions to get it. Like, the first result I looked at was: https://www.nycourts.gov/reporter/3dseries/2021/2021_07058.htm

Which has Docket No. 79 and 78. 

Should we use our post process hook to get the docket numbers?

Reply:
I think this was a failure of the previous scraper that I let slide.

Reply:
Yeah, I figured as much. Now that the can of worms is open though, seems like we should fix it?

Reply:
indeed.  we should.

Reply:
this should be done